File: csharp/main/PropertyDef.cs
using System;
using System.Collections.Generic;
using JetBrains.Annotations;

namespace Deltix.Luminary
{
	public abstract class PropertyDef<T> : ItemDef
	{
		protected PropertyDef([NotNull] T owner, [NotNull] String name, [NotNull] Type type, [CanBeNull, ItemNotNull] List<String> comments = null)
			: base(name, comments)
		{
			Type = type;
			Owner = owner;
		}

		[NotNull]
		public Type Type { get; }

		[NotNull]
		public T Owner { get; }
	}
}

File: java/src/main/java/deltix/luminary/PropertyDef.java
package deltix.luminary;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.List;

public abstract class PropertyDef<T extends ItemDef> extends ItemDef {
    private final Type type;
    private final T owner;

    public PropertyDef(@NotNull T owner, @NotNull String name, @NotNull Type type, @Nullable List<String> comments) {
        super(name, owner.getFullName() + "." + name, comments);
        this.type = type;
        this.owner = owner;
    }

    @NotNull
    public Type getType() {
        return type;
    }

    @NotNull
    public T getOwner() {
        return owner;
    }
}
